### DISCUSSION TITLE
Porting to MacOS and Linux

### DISCUSSION THREAD

**Discussion by SterlingSnead** [AntonyCorbett/OnlyM#414] [General]
Hello, I am interesting in assisting in porting over the MacOS and Linux whether that be a wine/crossover compatibility or by creating from scratch. I also know some other technical ones that would be able to assist. Would this be of interest for collaboration?

**Comment by AntonyCorbett** [comment]
@SterlingSnead I hope you get some offers.

**Comment by SterlingSnead** [comment]
Just tested again with Pop!_OS 21.04, Crossover Linux 20.0.0, & OnlyM 2.0.0.11. OnlyM now runs and will work displaying pictures. Resizing and scaling works. Video files will crash the app. Will see what I can do to assist. Until then the video player MPV will share in Zoom the video and not the title bar.

**Comment by radiant-forever** [comment]
JWLIB is buggy running on macOS using WINE and using Parallels and Windows.  I have not been able to get it to work reliably running on Windows using Parallels.

We have many that have gone over to using Apple gear over the years, with Zoom, the only reliable solution for macOS is OnlyM running in Coherence Mode in Parallels (Windows 10 in Coherence).  This works wonderfully, even though it takes a few minutes to extract all of the media from the website.  Note, OnlyM does not function correctly with two screens running Windows under parallels on a Mac when in full screen mode (at least that I've had the patience to tinker with).  Coherence Mode, however, just works.  Having multiple screens is a dream.  I have the Media folder as a favorite in MacOS finder, so I don't have to worry about navigating the Windows file management system (Parallels does a great job with mounting Windows Folders over to MacOS and Finder).

I get all the songs directly from JW.ORG and all the meeting media from wol.jw.org, then of course, public speakers provide their own media and friends may have additional media for talks.  Creating the playlist or queue list is a breeze appending the file names directly in the Media folder with 01, 02, 03, etc. or 01 PT, 01 WT, etc. depending on the meeting.  

I've found having a thumbnail of the song is super useful for the Chairman, simply taking a screen shot of the song paused at the sound title creates a perfect thumbnail to show prior to the song.  WT Study conductor also has us create slides of the Theme and Review Box, which we do using a screen shot.  Using OnlyM allows this very interactive approach and smooth transitions between these various types of media, both within the lessons, and ad hoc.

A native macOS version would be a wonderful addition... there is much more Apple gear than most realize... those that use, just learn to work around the legacy Windows environment (using Wine, Parallels when needed).

But I'll take this a step forward... by far the majority of friends no longer are using personal computers... it's all tablets and mobile devices.  This places a higher requirement for friends available to be used for Zoom AV support.  To have an OnlyM app in iOS would be fantastic and open up the field of AV possibilities over Zoom.  I do believe this is possible, as files can be extracted within iOS and saved to a folder as we do on the desktop.  Sharing JWLIB media via a mobile device does work adequately in a pinch, and JWLIB even supports a second screen in the App, if you setup with a cable.  But of course doing ad hoc media would be problematic and stressful to pull off seamlessly without a purpose built App designed to work within Zoom... my thought is now that Zoom supports Apps, an iOS/Android/Zoom App might be worth exploring.

In any case, thanks for this OnlyM software, it is much appreciated and we are able to pull off high quality seamless meetings displaying multiple media formats over zoom using OnlyM.  Friends really notice the difference between brothers using other methods, which shows all the mouse and cursor movements and play bars of the share screen versus the clean un-distracting display of OnlyM.  OnlyM presents a clean un-distracting  full screen view and completely takes the stress out of AV support once the media is in place.

Thank you!

**Comment by jlader** [comment]
As a possibility, if you have a PC running Windows 10 pro, you can use remote desktop to display the Windows display on your Mac (Free MS app for Mac). The PC doesn't need to be in the same room, just on the same network. This works best with a second monitor.  Then, from the your Mac, you can play from OnlyM and the JWLib and use your Mac.  Using the Zoom on your mac, you can share that screen, or a portion of it (I haven't gotten to trying this on a mac yet).
The fact that most friends are using tablets for Zoom doesn't matter.  Only those running the media or attendant on Zoom need a PC/Mac.  The Branch still recommends a PC with JWLib for media display for Halls and Regional Conventions.  OnlyM is probably the best media player to work in conjunction with this setup.

**Comment by dhaanpaa-lab0** [comment]
One possiblity if anthony wanted to he could use avaloniaui which is the cross platform equivilant to WPF.
